includes/FontFile.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Woda\WordPress\TwoStageFontsLoader;
+
+use InvalidArgumentException;
+
+final class FontFile
+{
+    /** @var string */
+    public $baseName;
+    /** @var string */
+    public $extension;
+    /** @var string */
+    public $path;
+    /** @var string */
+    public $url;
+
+    public function __construct(string $url)
+    {
+        $this->url = $url;
+        $this->path = $this->getUrlPath();
+        $this->baseName = $this->getBaseName();
+        $this->extension = $this->getExtension();
+    }
+
+    private function getBaseName(): string
+    {
+        return pathinfo($this->path, PATHINFO_BASENAME);
+    }
+
+    private function getExtension(): string
+    {
+        return pathinfo($this->path, PATHINFO_EXTENSION);
+    }
+
+    private function getUrlPath(): string
+    {
+        $urlPath = parse_url($this->url, PHP_URL_PATH);
+        return $urlPath ?: '';
+    }
+
+    public function hasValidExtension(): bool
+    {
+        $allowedExtensions = Config::loadConfigArray('valid_extensions');
```

Comment:
So ein Funktionsaufruf machts natürlich schwieriger zu testen. Über Alternativen können wir quatschen, ist vermutlich nicht so einfach da rauszubekommen :)

Reply:
Und eigentlich wichtiger, in Anlehnung an mein Kommentar weiter oben irgendwo:
Das sollte eigentlich schon beim construct passieren oder? Damit gar keine invaliden extensions angelegt werden können.